### DISCUSSION TITLE
Reolink Camera Plugin -

### DISCUSSION THREAD

**Discussion by Alias-ZA** [koush/scrypted#1176] [Q&A]
My setup:

Home Assistant OS on Intel NUC 
Scrypted 0.57.0
Reolink NVR (RLN16-410)
8 x RLC-811A
1 x Reolink Doorbell POE

All cameras are wired to the NVR with CAT 6. NVR is wired to network.  

I can load the RTMP stream of the Doorbell (Channel 0) into HA (Generic Camera) and a webhook for the still - this works perfectly and as expected. 

If I then go and add the next "normal" camera, it works as expected and on Scrypted I can see the still image and the feed is there if I click on the image. Looks to all be working fine. 

If I then go to add this camera (Channel 9) to a generic card - I get a timeout error. Opening the console of the camera in scrypted I see the below error:  [Rebroadcast Plugin]: RTMP main.bcs prebuffer session ended with error Error: ffmpeg killed before sdp could be parsed

Similar to this: https://github.com/koush/scrypted/issues/1146 - my password is simple though (instacam), all lowercase and no special char - works fine with the doorbell.

**Screenshots**

```
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session started
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session starting
stream url rtmp://10.0.4.5:1935/bcs/channel9_main.bcs?channel=9&stream=0&user=admin&password=instacam
[Rebroadcast Plugin]: -hide_banner -fflags +genpts -i rtmp:[REDACTED] -rtsp_transport tcp -vcodec copy -acodec copy -f rtsp rtsp://127.0.0.1:35049/d0e8ab99328f929c
debug event:
 [{"cmd":"GetMdState","code":0,"value":{"state":0}}]
[Rebroadcast Plugin]: rtmp://admin:instacam@10.0.4.5:1935/bcs/channel9_main.bcs?channel=9&stream=0&user=admin&password=instacam: End of file
[Rebroadcast Plugin]: ffmpeg exited
[Rebroadcast Plugin]: rebroadcast error Error: ffmpeg exited
[Rebroadcast Plugin]:     at ChildProcess.<anonymous> (/@scrypted/prebuffer-mixin/main.nodejs.js:1:76176)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:events:526:35)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:domain:489:12)
[Rebroadcast Plugin]:     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session ended with error Error: ffmpeg killed before sdp could be parsed
[Rebroadcast Plugin]:     at /@scrypted/prebuffer-mixin/main.nodejs.js:1:76493
[Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Rebroadcast Plugin]: Reolink Camera restarting prebuffer session in 5 seconds
debug event:
 [{"cmd":"GetMdState","code":0,"value":{"state":0}}]
debug event:
 [{"cmd":"GetMdState","code":0,"value":{"state":0}}]
debug event:
 [{"cmd":"GetMdState","code":0,"value":{"state":0}}]
debug event:
 [{"cmd":"GetMdState","code":0,"value":{"state":0}}]
```

Am I missing something stupid here?

**Comment by koush** [comment]
Would connect to camera directly if possible.

**Comment by Alias-ZA** [comment]
So not via the NVR?

**Comment by Alias-ZA** [comment]
Okay, I have taken all the cameras out of the NVR and they are now directly powered and plugged in via my Unifi Dream Machine Pro SE. 

I deleted all the entries in scrypted, restarted the HA server and re-added the camera's each with their own IP address. 

I get the same error. First camera works fine, the rest have the rebroadcast error. 

Any ideas?

**Comment by Alias-ZA** [comment]
Figured out the issue is related to the RLC-811A outputting 4k h265 which Scrypted doesn't like. 

I can downrate the stream to 1440p which works BUT I seem to lose significant FOV. This is likely a Reolink implementation thing but won't really work with my implementation. 

I do have an Unraid server with a Nvidia 1070 GPU - can I install Scrypted on docker and transcode the 4k h265 to h264? Will only be for streaming - NVR will do the recording.

**Reply by alexgorbatchev** [reply]
Check if you have the latest firmware, I had to upgrade my RLC-822A to get 4k streaming to work.

**Reply by gmcinalli** [reply]
@alexgorbatchev I've the latest firmware for my 811A, and the FOV is really reduced if I throw the stream resolution down to 2K.

**Comment by hyperbart** [comment]
I have a RLC810 and indeed, when I changed stream quality to 2560x1440p the camera works (i also had the error ffmpeg killed before sdp could be parsed).

Going to check if I need to do a FW upgrade

**Reply by hyperbart** [reply]
I have upgraded the firmware to the latest version (below) but issue i still there. Biggest problem is indeed the FOV that changes when changing the resolution.

![image](https://github.com/koush/scrypted/assets/2131310/71e2e5cd-018b-4fbf-9b28-a283ca9b78a0)

**Reply by jeremyward** [reply]
Did you have any luck with this?  I've got the same issue.

**Comment by hyperbart** [comment]
@koush i see this was originally an issue but you converted it to a discussion.
Isn't this a real issue that should be looked at via a bug report/issue?

**Comment by Matim1981** [comment]
My logs looks as bylow. Can some one help where is my mistake in configuration what and where should I change to have the streems running I am useing 4 Reolink P344 and 2 Duo 3 all conected to Reolink NVR. I'm a beginner so please be understanding ;-) 

rtsp://127.0.0.1:42415/870f3dd2360d28d6
[Rebroadcast Plugin]: [in#0 @ 0xaaaad6e2cda0] Error opening input: Input/output error
[Rebroadcast Plugin]: Error opening input file rtmp://192.168.1.33:1935/bcs/channel1_main.bcs?channel=1&stream=0&user=admin&password=*********.
[Rebroadcast Plugin]: Error opening input files: Input/output error
[Rebroadcast Plugin]: ffmpeg exited
[Rebroadcast Plugin]: Front door restarting prebuffer session in 5 seconds
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session ended with error Error: ffmpeg exited
[Rebroadcast Plugin]:     at ChildProcess.<anonymous> (/server/volume/plugins/@scrypted/prebuffer-mixin/zip/src/ffmpeg-rebroadcast.ts:197:30)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:events:530:35)
[Rebroadcast Plugin]:     at ChildProcess.emit (node:domain:489:12)
[Rebroadcast Plugin]:     at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session started
[Rebroadcast Plugin]: RTMP main.bcs prebuffer session starting
[Rebroadcast Plugin]: -hide_banner -fflags +genpts -i rtmp:[REDACTED] -rtsp_transport tcp -vcodec copy -acodec copy -f rtsp
